Make daemonset mountPropagation configurable, remove bidirectional default

jlubins opened the pull request:
This PR addresses issues #119 and #154 where the core-dump-handler causes excessive bind mounts due to bidirectional mount propagation. When pods restart frequently (e.g., due to memory limits being set too low), this has potential to create thousands of duplicate bind mounts, leading to:

- High I/O wait times
- Node performance degradation
- Potential node failures

The current implementation hardcodes mountPropagation: Bidirectional in the DaemonSet template. As noted in the Kubernetes [documentation](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) (see warning), bidirectional mount propagation can be dangerous and requires containers to properly clean up mounts on termination. Each pod restart doubles the number of mounts, quickly reaching system limits.


This PR makes mountPropagation configurable via Helm values and removes the bidirectional default. Users can now:
- Use the safer default (no mount propagation specified)
- Explicitly set bidirectional mode if needed for their use case
- Choose other propagation modes as appropriate

Commit:
Make daemonset mountPropagation configurable, remove bidirectional default

Signed-off-by: John Lubinski <john.lubinski@chicagotrading.com>

Commit:
Fix duplicated definition

Signed-off-by: John Lubinski <john.lubinski@chicagotrading.com>

jlubins commented:
Hi @No9, when you have a chance, could I trouble you to review this PR? It has seemingly bitten several users in the past, and prevents me from safely using core-dump-handler in quite a few places.

No9 commented:
@jlubins the lint is failing - can you update the PR to resolve this and I can merge

Commit:
Retrigger tests

Signed-off-by: John Lubinski <john.lubinski@chicagotrading.com>

Commit:
Update lint-test versions

Signed-off-by: John Lubinski <john.lubinski@chicagotrading.com>

jlubins commented:
Hey @No9, apologies for the delay. Do I need approval to get the lint test to run again?

With my latest commit, I've fixed the Python version issue that was causing the workflow to fail:
```
Run actions/setup-python@v2
Version 3.7 was not found in the local cache
Error: Version 3.7 with arch x64 not found
```

I updated the version per the README docs for setup-python [here](https://github.com/marketplace/actions/setup-python#basic-usage). It looks like chart-testing-action accepts [any latest minor version](https://github.com/helm/chart-testing-action?tab=readme-ov-file#example-workflow). I pinned to 3.13, but I can change it to '3.x' if you'd prefer the workflow to automatically use the latest Python 3 version.

jlubins commented:
Hi @No9, following up here again. It seems like I need approval for the workflows to run?

<img width="944" height="478" alt="image" src="https://github.com/user-attachments/assets/fb6def92-4ef9-48c9-bbc7-f54db29fde09" />

ProbstenHias commented:
@No9 we are actually just ran into a node failure because of this issue, so this fix would help us a lot.

ProbstenHias commented:
@jlubins please adjust the [values.schema.json](https://github.com/IBM/core-dump-handler/blob/main/charts/core-dump-handler/values.schema.json) file as well to include your new `mountPropagation` field.

Commit:
Update values.schema.json

Signed-off-by: John Lubinski <john.lubinski@chicagotrading.com>

davidxia commented:
Hi, I've also hit this issue on several K8s nodes. This fix will help me too if a maintainer can review again.

yahia-wk commented:
Another production hit for [#119](https://github.com/IBM/core-dump-handler/issues/119)/[#154](https://github.com/IBM/core-dump-handler/issues/154) here -  the `Bidirectional` default leaked host mounts and had to disable the handler to stop nodes accumulating them . could a maintainer (@No9 ) approve the CI run and take a look at merging this? Thanks!
